### ISSUE TITLE
Unable to move past "Applying Database Changes"

### ISSUE THREAD

**Issue by d3corator** [orangehrm/orangehrm#1622]
### OrangeHRM version

5.3

### Environment details

- OrangeHRM source: Github Releases
- Platform: Ubuntu 22.04
- PHP version: 8.1
- Database and version: 10.6.11-MariaDB-0ubuntu0.22.04.1
- Web server: Apache
- Browser: Firefox

### Describe the bug

When installing OrangeHRM it gives me an error when Applying Database changes, saying Request failed with status code 500. For more details check the error log in /src/log/installer.log file

Upon checking the log file I have found the following 
`[2023-02-15T12:33:46.035837+01:00] installer.ERROR: An exception occurred while executing a query: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'php_version' at row 1 [] []
[2023-02-15T12:33:46.035985+01:00] installer.ERROR: #0 /var/www/hrm/src/vendor/doctrine/dbal/src/Connection.php(1814): Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert() #1 /var/www/hrm/src/vendor/doctrine/dbal/src/Connection.php(1749): Doctrine\DBAL\Connection->handleDriverException() #2 /var/www/hrm/src/vendor/doctrine/dbal/src/Connection.php(1163): Doctrine\DBAL\Connection->convertExceptionDuringQuery() #3 /var/www/hrm/src/vendor/doctrine/dbal/src/Query/QueryBuilder.php(324): Doctrine\DBAL\Connection->executeStatement() #4 /var/www/hrm/installer/Util/MigrationHelper.php(95): Doctrine\DBAL\Query\QueryBuilder->executeStatement() #5 /var/www/hrm/installer/Util/AppSetupUtility.php(523): OrangeHRM\Installer\Util\MigrationHelper->logMigrationStarted() #6 /var/www/hrm/installer/Util/AppSetupUtility.php(512): OrangeHRM\Installer\Util\AppSetupUtility->_runMigration() #7 /var/www/hrm/installer/Controller/Upgrader/Api/MigrationAPI.php(50): OrangeHRM\Installer\Util\AppSetupUtility->runMigrationFor() #8 /var/www/hrm/installer/Controller/AbstractInstallerRestController.php(57): OrangeHRM\Installer\Controller\Upgrader\Api\MigrationAPI->handlePost() #9 /var/www/hrm/installer/Controller/AbstractInstallerController.php(85): OrangeHRM\Installer\Controller\AbstractInstallerRestController->execute() #10 /var/www/hrm/src/vendor/symfony/http-kernel/HttpKernel.php(152): OrangeHRM\Installer\Controller\AbstractInstallerController->handle() #11 /var/www/hrm/src/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw() #12 /var/www/hrm/installer/Framework/HttpKernel.php(244): Symfony\Component\HttpKernel\HttpKernel->handle() #13 /var/www/hrm/installer/index.php(66): OrangeHRM\Installer\Framework\HttpKernel->handleRequest() #14 {main} [] []`

### To reproduce

Not Sure

### Expected behavior

Installation should not halt

### Relevant log output

```shell
[2023-02-15T12:33:46.035837+01:00] installer.ERROR: An exception occurred while executing a query: SQLSTATE[22001]: String data, right truncated: 1406 Data too long for column 'php_version' at row 1 [] []
[2023-02-15T12:33:46.035985+01:00] installer.ERROR: #0 /var/www/hrm/src/vendor/doctrine/dbal/src/Connection.php(1814): Doctrine\DBAL\Driver\API\MySQL\ExceptionConverter->convert() #1 /var/www/hrm/src/vendor/doctrine/dbal/src/Connection.php(1749): Doctrine\DBAL\Connection->handleDriverException() #2 /var/www/hrm/src/vendor/doctrine/dbal/src/Connection.php(1163): Doctrine\DBAL\Connection->convertExceptionDuringQuery() #3 /var/www/hrm/src/vendor/doctrine/dbal/src/Query/QueryBuilder.php(324): Doctrine\DBAL\Connection->executeStatement() #4 /var/www/hrm/installer/Util/MigrationHelper.php(95): Doctrine\DBAL\Query\QueryBuilder->executeStatement() #5 /var/www/hrm/installer/Util/AppSetupUtility.php(523): OrangeHRM\Installer\Util\MigrationHelper->logMigrationStarted() #6 /var/www/hrm/installer/Util/AppSetupUtility.php(512): OrangeHRM\Installer\Util\AppSetupUtility->_runMigration() #7 /var/www/hrm/installer/Controller/Upgrader/Api/MigrationAPI.php(50): OrangeHRM\Installer\Util\AppSetupUtility->runMigrationFor() #8 /var/www/hrm/installer/Controller/AbstractInstallerRestController.php(57): OrangeHRM\Installer\Controller\Upgrader\Api\MigrationAPI->handlePost() #9 /var/www/hrm/installer/Controller/AbstractInstallerController.php(85): OrangeHRM\Installer\Controller\AbstractInstallerRestController->execute() #10 /var/www/hrm/src/vendor/symfony/http-kernel/HttpKernel.php(152): OrangeHRM\Installer\Controller\AbstractInstallerController->handle() #11 /var/www/hrm/src/vendor/symfony/http-kernel/HttpKernel.php(74): Symfony\Component\HttpKernel\HttpKernel->handleRaw() #12 /var/www/hrm/installer/Framework/HttpKernel.php(244): Symfony\Component\HttpKernel\HttpKernel->handle() #13 /var/www/hrm/installer/index.php(66): OrangeHRM\Installer\Framework\HttpKernel->handleRequest() #14 {main} [] []
```

**Comment by d3corator** [comment]
FYI: Just installed 5.2 successfully so this seems to be an issue with 5.3

**Comment by MrTrinax** [comment]
Seems the issue occurs due to the column "php_version" in the "ohrm_migration_log" table being a type varchar(10).
In my case the instance detected a PHP version "8.1.2-1ubuntu2.11" which was too long for the "php_version" column.

To fix this I just changed the PHP version.

**Comment by cakkineni** [comment]
> Seems the issue occurs due to the column "php_version" in the "ohrm_migration_log" table being a type varchar(10). In my case the instance detected a PHP version "8.1.2-1ubuntu2.11" which was too long for the "php_version" column.
> 
> To fix this I just changed the PHP version.

How did you change the php_version; did you install a new version?

**Comment by cakkineni** [comment]
> FYI: Just installed 5.2 successfully so this seems to be an issue with 5.3

Using php 5.2 gives an error on mysql version should be between 5.2 & 5.7, but current version is 8.2 .. did you end up installing one of the older versions?

**Comment by gwijayas** [comment]
I have the same problem. 
- OrangeHRM 5.3
- PHP 7.4.33, PHP 8.1
- MySQL 5.7

**Comment by gwijayas** [comment]
Got the sampe problem.
- OrangeHRM 5.3
- PHP 7.4.33/8.1.17
- MySQL 5.7

Log:
```[2023-03-23T21:21:06.575925+07:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"5.19.0-35-generic","version_info":"#36~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Feb 17 15:17:25 UTC 2"},"php":{"version":"7.4.33"},"mysql":{"client_version":"mysqlnd 7.4.33","server_version":"5.7.36","conn_type":"127.0.0.1 via TCP\/IP"},"server":"Apache","ohrm":{"version":"5.3"}} [] []
[2023-03-23T21:21:06.682206+07:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"http://hrm.example.com/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2023-03-23T21:21:06.682578+07:00] installer.INFO: {"version":"4.10.1"} [] []
[2023-03-23T21:21:06.683850+07:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"5.19.0-35-generic","version_info":"#36~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Feb 17 15:17:25 UTC 2"},"php":{"version":"7.4.33"},"mysql":{"client_version":"mysqlnd 7.4.33","server_version":"5.7.36","conn_type":"127.0.0.1 via TCP\/IP"},"server":"Apache","ohrm":{"version":"5.3"}} [] []
[2023-03-23T21:21:06.792408+07:00] installer.INFO: Matched route "{route}". {"route":"api_installer_migration","route_parameters":{"_controller":"OrangeHRM\\Installer\\Controller\\Installer\\Api\\MigrationAPI::handle","_route":"api_installer_migration"},"request_uri":"http://hrm.example.com/installer/index.php/installer/api/installation/migration","method":"POST"} []
[2023-03-23T21:21:06.792792+07:00] installer.INFO: {"version":"5.0"} [] []
[2023-03-23T21:21:06.794155+07:00] installer.INFO: Instance details: {"os":{"os":"Linux","release_name":"5.19.0-35-generic","version_info":"#36~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Fri Feb 17 15:17:25 UTC 2"},"php":{"version":"7.4.33"},"mysql":{"client_version":"mysqlnd 7.4.33","server_version":"5.7.36","conn_type":"127.0.0.1 via TCP\/IP"},"server":"Apache","ohrm":{"version":"5.3"}} [] []
```

my.cnf:
```
sql_mode="STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION"
```

**Comment by gwijayas** [comment]
> > > FYI: Just installed 5.2 successfully so this seems to be an issue with 5.3
> > 
> > 
> > Using php 5.2 gives an error on mysql version should be between 5.2 & 5.7, but current version is 8.2 .. did you end up installing one of the older versions?
> 
> I ended up installing PHP 8.2 instead of 8.1 because of the 8.1 version number string being to long for the "php_version" column in the SQL database. You can probably get away using any supported PHP version as long as the version number string isn't too long.

What Linux distro do you use? I am using Ubuntu 22.04, test with php 7.4/8.1, from [PPA](https://launchpad.net/~ondrej/+archive/ubuntu/php) got the similar problem, eventhough the log is quite different. And where did you see "php_version" thing?

**Comment by RajithaKumara** [comment]
Hi,
Seems this is a bug in the installer.
Please find the attached patch, and replace it with `installer/Util/MigrationHelper.php`
[MigrationHelper.zip](https://github.com/orangehrm/orangehrm/files/11190474/MigrationHelper.zip)

**Comment by kavithagurusamy** [comment]
This issue is fixed from OrangeHRM 5.4 version

**Comment by zeguonline** [comment]
Nothing is fixed, can't install version 5.7
